server/apps/publish/archive_publish.py

```diff
@@ -466,15 +466,18 @@ def filter_subscribers(self, doc, subscribers, target_media_type):
                     continue
 
             if doc.get('targeted_for'):
-                matching_target = [t for t in doc.get('targeted_for')
-                                   if t['name'] == subscriber.get('subscriber_type', '') or
-                                   t['name'] == subscriber.get('geo_restrictions', '')]
+                found_match = [t for t in doc['targeted_for'] if t['name'] == subscriber.get('subscriber_type', '')]
 
-                if len(matching_target) > 0 and matching_target[0]['allow'] is False:
+                if len(found_match) == 0 and subscriber.get('geo_restrictions'):
+                    found_match = [t for t in doc['targeted_for'] if t['name'] == subscriber['geo_restrictions']]
+                    if len(found_match) == 0 or (len(found_match) > 0 and found_match[0]['allow'] is False):
```

Comment:
The `len(found_match) > 0` check is redundant as it always evaluates to `True` if checking comes to this point.